packages/wonder-blocks-dropdown/components/multi-select-menu.md

```diff
@@ -0,0 +1,241 @@
+The multi select menu allows the selection of multiple items. Clients are
+responsible for keeping track of the selected items in the menu.
+
+TODO(sophie): keyboard navigation and dismissal
+
+The first menu starts with nothing selected and has no selection shortcuts.
+There is also a disabled item, and this menu has a set width.
+
+The second menu starts with one item selected and has selection shortcuts for
+select all and select none. The second menu does not have a predefined placeholder.
+
+The third menu is disabled.
+
+```js
+const React = require("react");
+const {View} = require("@khanacademy/wonder-blocks-core");
+const {StyleSheet} = require("aphrodite");
+
+const styles = StyleSheet.create({
+    row: {
+        flexDirection: "row",
+        height: 350,
+    },
+    strutLike: {
+        width: 8,
+    }
+});
+
+class ExampleNoneSelected extends React.Component {
+    constructor() {
+        super();
+        this.state = {
+            selectedItems: [],
+        };
+    }
+
+    handleChanges(changes) {
+        console.log("changes happened!");
+        const {selectedItems} = this.state;
+        changes.forEach((change) => {
+            if (change.selected) {
+                selectedItems.push(change.value);
+            } else {
+                selectedItems.splice(selectedItems.indexOf(change.value), 1);
```

Comment:
If `change` was also an array then this would become:
```
this.setState({
   selectedItems: changes,
});
```

Reply:
This will also simplify some of the other methods in `MultiSelectMenu`.

Reply:
I'm not sure what you mean here. `changes` is an array of items that have changed states. Do you mean that `changes` should return an array of all the selected items?

Reply:
Yes, if it's just an array of selected items there's a lot of code that can be simplified.  Besides, this is what's being passed as a prop to `MultiSelectMenu`.